researchstack-sdk/src/test/java/org/sagebionetworks/bridge/researchstack/factory/ArchiveFileFactoryTest.java

```diff
@@ -0,0 +1,104 @@
+package org.sagebionetworks.bridge.researchstack.factory;
+
+import org.junit.Test;
+import org.researchstack.backbone.answerformat.AnswerFormat;
+import org.researchstack.backbone.answerformat.ChoiceAnswerFormat;
+import org.researchstack.backbone.result.FileResult;
+import org.researchstack.backbone.result.StepResult;
+import org.sagebionetworks.bridge.data.ByteSourceArchiveFile;
+import org.sagebionetworks.bridge.data.JsonArchiveFile;
+import org.sagebionetworks.bridge.researchstack.survey.SurveyAnswer;
+import org.sagebionetworks.bridge.rest.RestUtils;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by liujoshua on 11/10/2017.
+ */
+public class ArchiveFileFactoryTest {
+    private ArchiveFileFactory archiveFileFactory = ArchiveFileFactory.INSTANCE;
+
+    @Test
+    public void toArchiveFile_forStepResult() throws Exception {
+        StepResult<List<String>> stepResultMock = createStepResultMock(
+                new ChoiceAnswerFormat(AnswerFormat.ChoiceAnswerStyle.SingleChoice)
+        );
+
+        List<String> result = Arrays.asList("Result1");
+        when(stepResultMock.getResult()).thenReturn(result);
+
+        SurveyAnswer.ChoiceSurveyAnswer answer = createAnswer(stepResultMock,
+                SurveyAnswer.ChoiceSurveyAnswer.class);
+
+        JsonArchiveFile file = archiveFileFactory.fromStepResult(stepResultMock);
+
+        String jsonString = file.getByteSource().asCharSource(StandardCharsets.UTF_8).read();
+
+        SurveyAnswer.ChoiceSurveyAnswer deserializedAnswer =
+                RestUtils.GSON.fromJson(jsonString, SurveyAnswer.ChoiceSurveyAnswer.class);
+
+        assertEquals(answer.item, deserializedAnswer.item);
+        assertEquals(answer.startDate, deserializedAnswer.startDate);
+        assertEquals(answer.endDate, deserializedAnswer.endDate);
+        assertEquals(answer.questionType, deserializedAnswer.questionType);
+        assertEquals(answer.questionTypeName, deserializedAnswer.questionTypeName);
+        assertEquals(answer.choiceAnswers, deserializedAnswer.choiceAnswers);
```

Comment:
Can we explicitly test that deserializedAnswer.choiceAnswers contains "Result1". I'm concerned that if there's somehow another bug preventing the answer from being wired through, we may end up just checking that null equals null in this test.